novawallet/Modules/Notifications/Management/NotificationsManagementPresenter.swift

```diff
@@ -244,8 +265,40 @@ extension NotificationsManagementPresenter: NotificationsManagementPresenterProt
         updateView()
     }
 
+    func changeMultisigSettings(result: MultisigNotificationsModel) {
+        modifiedSettings = modifiedSettings?.with {
+            $0.newMultisig = result.signatureRequested ? .all : nil
+            $0.multisigApproval = result.signedBySignatory ? .all : nil
+            $0.multisigExecuted = result.transactionExecuted ? .all : nil
+            $0.multisigCanceled = result.transactionRejected ? .all : nil
+        }
+
+        disableNotificationIfNeeded()
+        updateView()
+    }
+
     func changeWalletsSettings(wallets: [Web3Alert.LocalWallet]) {
-        modifiedSettings = modifiedSettings?.with(wallets: wallets)
+        guard let allWallets else { return }
+
+        let allWalletsMap = allWallets.reduceToDict()
+
+        let selectedMultisigs = wallets
+            .compactMap { allWalletsMap[$0.metaId] }
+            .filter { $0.type == .multisig }
+
+        if selectedMultisigs.isEmpty {
+            modifiedSettings = modifiedSettings?
+                .with(wallets: wallets)
+                .with {
+                    $0.newMultisig = nil
```

Comment:
Is this agreed logic that if no multisig selected then settings are cleared? Worth to sync with Android